### PR TITLE
Bulk actions for data node

### LINKED ISSUE #17732: Bulk actions for Data Node restart and removal
Create bulk actions for sequential data node restart and removal

### PR DESCRIPTION
**Pull request by moesterheld**
## Description
Provides functionality for bulk removal/start/stop of data nodes.

Screenshot 2024-02-02 at 15 41 06

## Motivation and Context
resolves #17732 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by todvora — APPROVED** [review]
LGTM!

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 1]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+export const bulkRemoveDataNode = async (entity_ids: string[], selectBackFailedEntities: (entity_ids: string[]) => void) => {
+  try {
+    const { failures, successfully_performed } = await fetch('POST', qualifyUrl('/datanode/bulk_remove'), { entity_ids });
+
+    if (failures?.length) {
+      selectBackFailedEntities(failures.map(({ entity_id }) => entity_id));
+    }
+
+    if (failures?.length === entity_ids.length) {
+      UserNotification.error(`Removing Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not remove Datanodes.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:35]:
```suggestion
      UserNotification.error(`Removing Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not remove Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 2]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+    }
+
+    if (successfully_performed) {
+      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} removed successfully`);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:39]:
```suggestion
      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} removed successfully.`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 3]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+    }
+  } catch (errorThrown) {
+    UserNotification.error(`Removing Data Node failed with status: ${errorThrown}`, 'Could not remove some Datanodes.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:42]:
```suggestion
    UserNotification.error(`Removing Data Node failed with status: ${errorThrown}`, 'Could not remove some Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 4]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+export const bulkStartDataNode = async (entity_ids: string[], selectBackFailedEntities: (entity_ids: string[]) => void) => {
+  try {
+    const { failures, successfully_performed } = await fetch('POST', qualifyUrl('/datanode/bulk_start'), { entity_ids });
+
+    if (failures?.length) {
+      selectBackFailedEntities(failures.map(({ entity_id }) => entity_id));
+    }
+
+    if (failures?.length === entity_ids.length) {
+      UserNotification.error(`Starting Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not start Datanodes.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:55]:
```suggestion
      UserNotification.error(`Starting Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not start Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 5]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+    }
+
+    if (successfully_performed) {
+      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} started successfully`);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:59]:
```suggestion
      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} started successfully.`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 6]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+    }
+  } catch (errorThrown) {
+    UserNotification.error(`Starting Data Node failed with status: ${errorThrown}`, 'Could not start Datanodes.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:62]:
```suggestion
    UserNotification.error(`Starting Data Node failed with status: ${errorThrown}`, 'Could not start Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 7]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+export const bulkStopDataNode = async (entity_ids: string[], selectBackFailedEntities: (entity_ids: string[]) => void) => {
+  try {
+    const { failures, successfully_performed } = await fetch('POST', qualifyUrl('/datanode/bulk_stop'), { entity_ids });
+
+    if (failures?.length) {
+      selectBackFailedEntities(failures.map(({ entity_id }) => entity_id));
+    }
+
+    if (failures?.length === entity_ids.length) {
+      UserNotification.error(`Stopping Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not stop Datanodes.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:75]:
```suggestion
      UserNotification.error(`Stopping Data Node failed with status: ${JSON.stringify(failures)}`, 'Could not stop Data Nodes.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 8]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+    }
+
+    if (successfully_performed) {
+      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} stopped successfully`);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:79]:
```suggestion
      UserNotification.success(`${successfully_performed} Data Node${successfully_performed > 1 ? 's' : ''} stopped successfully.`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 9]
```diff
@@ -23,6 +23,66 @@ import UserNotification from 'util/UserNotification';
...
+    }
+  } catch (errorThrown) {
+    UserNotification.error(`Stopping Data Node failed with status: ${errorThrown}`, 'Could not stop Datanodes.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:82]:
```suggestion
    UserNotification.error(`Stopping Data Node failed with status: ${errorThrown}`, 'Could not stop Data Nodes.');
```

**Review by grotlue — APPROVED** [review]
Looks good, just some small typo fixes. Please change before merge. I will approve already as I don't want to block this PR because of the typos.

**Pull request merged**